docs/building-on-etherlink/transactions.md

```diff
@@ -414,3 +414,148 @@ The response includes information about the matching events:
 ```
 
 For other filters, see [`eth_getLogs`](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getstorageat) in the Ethereum JSON-RPC API reference.
+
+## Tracing transactions
+
+Etherlink supports the `debug_traceTransaction` and `debug_traceBlockByNumber` RPC endpoints, which provide debugging information about transactions and blocks.
+
+To get debugging information about a transaction with `debug_traceTransaction`, pass the hash of the transaction, as in this example:
+
+```bash
+curl --request POST \
+     --url https://node.ghostnet.etherlink.com \
+     --header 'accept: application/json' \
+     --header 'content-type: application/json' \
+     --data '
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "debug_traceTransaction",
+  "params": [
+    "0x7b4c5cd4816aa5b529e11f9f5799d6a0f4b04ac53496688fef35c5762c5f9601"
+  ]
+}
+'
+```
+
+The response includes data about the transaction, including its status, gas cost, and return value.
+It also includes the structured logs from the EVM execution of the transaction.
+
+You can also get this information via ethers.js, as in this example:
+
+```javascript
+async function sendTransaction() {
+
+  // Send the transaction
+  const simpleTransaction = await simpleContract.set(5);
+  console.log("Transaction hash:", simpleTransaction.hash);
+
+  // Wait for the transaction to be confirmed
+  await simpleTransaction.wait();
+  console.log("Transaction confirmed!");
+
+  // Get trace information and print the top level of data
+  const trace = await provider.send("debug_traceTransaction", [simpleTransaction.hash]);
+  console.log("Transaction trace output:");
+  console.dir(trace, { depth: 1 });
+}
+```
+
+The result looks like this:
+
+```
+Transaction hash: 0x11231008cfe87e107cadca34136da439c4e980b27af7c6c84e4b44c93aab45a0
+Transaction confirmed!
+Trace output:
+{
+  gas: '21183',
+  failed: false,
+  returnValue: '0x',
+  structLogs: [
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object],
+    [Object], [Object], [Object]
+  ]
+}
+```
+
+Similarly, you can use the `debug_traceBlockByNumber` endpoint to get information about a block by passing its number in hexadecimal, as in the following example:
+
+```bash
+curl --request POST \
+     --url https://node.ghostnet.etherlink.com \
+     --header 'accept: application/json' \
+     --header 'content-type: application/json' \
+     --data '
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "debug_traceBlockByNumber",
+  "params": [
+    "0x011b1ac0",
+    {"tracer": "callTracer"}
+  ]
+}
+'
+```
+
+:::note
+
+Etherlink supports only the `callTracer` tracer for this endpoint, and only for blocks after the Bifröst upgrade.
```

Comment:
It could also be worth noting `debug_traceTransaction` supports only `structLogger` and `callTracer`, the former being the default (as in [Geth](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#debugtracetransaction)). A reference to [Geth documentation](https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers) could also be useful, so that users can have a look at the full specification of these two tracers.

Reply:
Ok, mentioned that.

Reply:
Thank you!